src/ConnectionPool.jl

```diff
@@ -488,6 +488,15 @@ end
 function getconnection(::Type{SSLContext},
                        host::AbstractString,
                        port::AbstractString;
+                       kw...)::SSLContext
+
+    port = isempty(port) ? "443" : port
+    @debug 2 "SSL connect: $host:$port..."
+    tcp = getconnection(TCPSocket, host, port; kw...)
+    return sslconnection(tcp, host; kw...)
+end
```

Comment:
Here I've split the second part of `getconnection` into a new `sslconnection` function that does the MbedTLS hookup. This is needed so that `sslupgrade` (below) can hook MbedTLS to an existing TCP connection after a `CONNECT` to a proxy.
Aside from being split in two, `getconnection` should do exactly what it did before.